p2p/src/sync/mod.rs

```diff
@@ -61,25 +62,32 @@ const HEADER_LIMIT: usize = 2000;
 /// peer it's connected to and actively keep track of the peer's state.
 pub struct BlockSyncManager<T: NetworkingService> {
     /// The chain configuration.
-    chain_config: Arc<ChainConfig>,
+    _chain_config: Arc<ChainConfig>,
 
     /// The p2p configuration.
-    _p2p_config: Arc<P2pConfig>,
+    p2p_config: Arc<P2pConfig>,
 
-    /// Handle for sending/receiving syncing events
-    peer_sync_handle: T::SyncingMessagingHandle,
+    /// A handle for sending/receiving syncing events.
+    messaging_handle: T::SyncingMessagingHandle,
 
-    /// RX channel for receiving control events
-    rx_sync: mpsc::UnboundedReceiver<SyncControlEvent<T>>,
+    /// A receiver for connect/disconnect events.
+    peer_event_receiver: mpsc::UnboundedReceiver<SyncControlEvent<T>>,
 
     /// A sender for the peer manager events.
-    tx_peer_manager: mpsc::UnboundedSender<PeerManagerEvent<T>>,
+    peer_manager_sender: mpsc::UnboundedSender<PeerManagerEvent<T>>,
 
-    /// Hashmap of connected peers
-    peers: HashMap<T::PeerId, peer::PeerContext<T>>,
+    /// A mapping from a peer identifier to the context for every connected peer.
+    peers: HashMap<T::PeerId, PeerContext>,
 
-    /// Subsystem handle to Chainstate
+    /// A handle to the chainstate subsystem.
     chainstate_handle: subsystem::Handle<Box<dyn chainstate_interface::ChainstateInterface>>,
+
+    /// A queue of the blocks requested by peers.
+    ///
+    /// The block identifiers are added to the queue as a result of BlockListRequest processing
+    /// and removed either after sending a response or when the peer is disconnected. A number of
+    /// blocks is limited by `P2pConfig::requested_blocks_limit` per peer.
+    blocks_queue: VecDeque<(T::PeerId, Id<Block>)>,
```

Comment:
What is the point of `blocks_queue`?
I think it's pretty useless because there is no backpressure.
This line will most likely be selected first if `blocks_queue` is not empty, and the code will work much the same as without using `blocks_queue` at all.

```
tokio::select! {
...
    (peer, block) = async { self.blocks_queue.pop_front().expect("The block queue is empty") }, if !self.blocks_queue.is_empty() => {
        let res = self.send_block(peer, block).await;
        self.handle_result(peer, res).await?;
    }
}
```

Reply:
@stanislav-tkach explained that it's needed to allow sync manager to receive messages without blocking the event loop for too long.